spectator_mode/init.lua

```diff
@@ -1,5 +1,53 @@
 local spectators = {}
 
+--- Join messages aren't hidden in this mod because those are handled by the CTF game
+function minetest.send_leave_message(player_name, timed_out)
+	if not minetest.check_player_privs(player_name, {spectate = true}) then
+		minetest.chat_send_all("*** " .. player_name .. " left the game.")
+	end
+end
+
+local function privs_of(name, privs)
+	if not privs then
+		privs = minetest.get_player_privs(name)
+	end
+	local privstr = minetest.privs_to_string(privs, ", ")
+	if privstr == "" then
+		return name .. " does not have any privileges."
+	else
+		return "Privileges of " .. name .. ":" .. privstr
+	end
+end
```

Comment:
```suggestion
local function privs_of(name, privs)
	if not privs then
		privs = minetest.get_player_privs(name)
	end
	local privstr = minetest.privs_to_string(privs, ", ")
	if privstr == "" then
		return name .. " does not have any privileges."
	else
		return "Privileges of " .. name .. ":" .. privstr:gsub(" ?spectate,", ""):gsub(", spectate$", "")
	end
end
```

Does this work? It should remove `spectate` from the priv string, taking commas into account
(http://lua-users.org/wiki/PatternsTutorial btw)

Reply:
Ban, kick and other "staff privs" will still be shown that way I guess. Checking for 'spectate' and 'secret' makes sure none of these would be shown